### PR TITLE
Update Apps docker dev environment to use Apps plugin 1.2 and server 7.6

### PR DESCRIPTION
**Pull request by javaguirre**
#### Summary

Update Apps 1.2.0 MM 7.4.0

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-43196

### CODE REVIEW

**Review by levb — APPROVED** [review]

**Review by azigler — APPROVED** [review]
@javaguirre Thanks for doing this -- can we go ahead and merge?

File: docker-compose.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ networks:
     name: "mattermost-apps-dev"
 services:
   mattermost:
-    image: "mattermost/mattermost-enterprise-edition:7.2" # https://hub.docker.com/r/mattermost/mattermost-enterprise-edition/tags
+    image: "mattermost/mattermost-enterprise-edition:7.4.0" # https://hub.docker.com/r/mattermost/mattermost-enterprise-edition/tags
```
Comment (hanzei) [inline docker-compose.yml:7]: Should we go to 7.5 directly?

**Review by hanzei — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (javaguirre) [reply docker-compose.yml:7]: sure, why not?

**Comment by javaguirre** [comment]
> @javaguirre Thanks for doing this -- can we go ahead and merge?

we haven't cut the v1.2 of the apps framework yet, that's why I put it as Draft. I'd like to test it when we do, shouldn't be a problem, but just in case. :-)

**Review by hanzei — CHANGES_REQUESTED** [review]
Two question:

1. Should other version numbers like https://github.com/mattermost/mattermost-app-examples/blob/be70ed24d88cbd1f0cd333e0bb0fcdfdbb8405d9/golang/webhooks/hello.go#L27 also get updated?
2. Should the `go.mod` files get updated to import `v1.2.0` of github.com/mattermost/mattermost-plugin-apps?

**Comment by javaguirre** [comment]
> Two question:
> 
> 1. Should other version numbers like https://github.com/mattermost/mattermost-app-examples/blob/be70ed24d88cbd1f0cd333e0bb0fcdfdbb8405d9/golang/webhooks/hello.go#L27
>     also get updated?
> 2. Should the `go.mod` files get updated to import `v1.2.0` of github.com/mattermost/mattermost-plugin-apps?

Yes, let's do that once we cut the release 👌

**Comment by dsharma522** [comment]
Please consider mm-server version to [7.1.5](https://github.com/mattermost/mattermost-server/releases/tag/v7.1.5)

**Comment by hanzei** [comment]
@dsharma522 Why should the version to changed to 7.1.5, then 7.6 is the newer release?

**Comment by levb** [comment]
@dsharma522 *7.5* is the correct version, and that's what's in the PR.

**Comment by javaguirre** [comment]
I'm getting an error with these versions, checking out what is happening.
Screenshot 2023-01-10 at 10 45 06

**Comment by javaguirre** [comment]
Screenshot 2023-01-10 at 10 53 02

**Comment by javaguirre** [comment]
I see we need to have it in the marketplace to finish testing this.

Screenshot 2023-01-10 at 12 10 44

**Comment by azigler** [comment]
@javaguirre Are we ready to revisit this PR now?

**Comment by javaguirre** [comment]
Hey @azigler ! Waiting on this https://github.com/mattermost/mattermost-marketplace/pull/331

**Review by hanzei — CHANGES_REQUESTED** [review]
The version number in https://github.com/mattermost/mattermost-app-examples/blob/be70ed24d88cbd1f0cd333e0bb0fcdfdbb8405d9/golang/serverless/manifest.json#L3 also should get updated

**Comment by javaguirre** [comment]
Done

**Review by hanzei — APPROVED** [review]
LGTM :rocket:

**Pull request merged**